### PR TITLE
Add mkldnn sigmoid operator

### PR DESCRIPTION
**Pull request by bddppq**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#20820 Add mkldnn sigmoid operator**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15455866/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #20800 Enable torch.jit.trace for mkldnn modules&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15447892/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #20799 Add support for save and load mkldnn modules&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15447891/)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/20820

Differential Revision: [D15455866](https://our.internmc.facebook.com/intern/diff/D15455866/)

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -1600,6 +1600,11 @@
 
 - func: celu_(Tensor(a!) self, Scalar alpha=1.0) -> Tensor(a!)
 
+- func: mkldnn_sigmoid(Tensor self) -> Tensor
+  dispatch:
+    MkldnnCPU: mkldnn_sigmoid
```
Comment (dzhulgakov) [inline aten/src/ATen/native/native_functions.yaml]: can't you put this dispatch directly on `sigmoid`? It does have an entry in Derivatives.yaml so it should work as a direct dispatch already happens there

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -1600,6 +1600,11 @@
...
+  requires_tensor: True
+
 - func: sigmoid(Tensor self) -> Tensor
   variants: function, method
```
Comment (dzhulgakov) [inline aten/src/ATen/native/native_functions.yaml]: we probably want to make in-place sigmoid below work too (`sigmoid_(Tensor(a!) self) -> Tensor(a!)`)

Comment (bddppq) [reply aten/src/ATen/native/native_functions.yaml]: currently in-place doesn't work in mkldnn tensors

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (bddppq) [reply aten/src/ATen/native/native_functions.yaml]: Sure will try

**Review by dzhulgakov — APPROVED** [review]

**Pull request closed without merging**